src/node/test/channels.cpp

```diff
@@ -18,7 +18,7 @@
 threading::ThreadMessaging threading::ThreadMessaging::thread_messaging;
 std::atomic<uint16_t> threading::ThreadMessaging::thread_count = 0;
 
-constexpr auto buffer_size = 1024 * 16;
+constexpr auto buffer_size = 1024 * 8;
```

Comment:
See the comment below about `ringbuffer::read()`. I was getting some very strange failures in this unit test, until I worked out it was just missing a message because it had reached the end of the buffer, and needed `read()` to be called again. So we now call it twice, and prove that's sane by reducing `buffer_size` (and seeing that other tests now fail without the double-read).